Explorer/Assets/DCL/LOD/Components/SceneLODInfo.cs

```diff
@@ -1,55 +1,114 @@
 using ECS.StreamableLoading.Common;
-using System;
-using DCL.AvatarRendering.AvatarShape.Rendering.TextureArray;
-using DCL.Profiling;
 using UnityEngine;
 using Utility;
 
 namespace DCL.LOD.Components
 {
     public struct SceneLODInfo
     {
-        public byte CurrentLODLevel;
-        public LODAsset CurrentLOD;
-        public LODAsset CurrentVisibleLOD;
         public AssetPromise<AssetBundleData, GetAssetBundleIntention> CurrentLODPromise;
-        public bool IsDirty;
+        public byte CurrentLODLevelPromise;
+
+        public string id;
+        public LODCacheInfo metadata;
         
         public void Dispose(World world)
         {
             CurrentLODPromise.ForgetLoading(world);
-            if (CurrentVisibleLOD != null && !CurrentVisibleLOD.LodKey.Equals(CurrentLOD))
-                CurrentVisibleLOD.Release();
-            CurrentLOD?.Release();
         }
 
-        public static SceneLODInfo Create() =>
-            new()
+        public static SceneLODInfo Create()
+        {
+            return new SceneLODInfo
             {
-                CurrentLODLevel = byte.MaxValue
+                CurrentLODLevelPromise = byte.MaxValue
             };
+        }
 
-        public void SetCurrentLOD(LODAsset newLod)
+        public void AddFailedLOD()
         {
-            CurrentLOD = newLod;
-            UpdateCurrentVisibleLOD();
+            metadata.FailedLODs = SceneLODInfoUtils.SetLODResult(metadata.FailedLODs, CurrentLODLevelPromise);
+            CurrentLODLevelPromise = byte.MaxValue;
         }
 
-        public void UpdateCurrentVisibleLOD()
+
+        public void AddSuccessLOD(GameObject instantiatedLOD, LODAsset lodAsset, float defaultFOV, float defaultLodBias, int loadingDistance)
         {
-            if (CurrentLOD?.State == LODAsset.LOD_STATE.SUCCESS)
+            metadata.SuccessfullLODs = SceneLODInfoUtils.SetLODResult(metadata.SuccessfullLODs, CurrentLODLevelPromise);
+            int loadedLODAmount = SceneLODInfoUtils.CountLOD(metadata.SuccessfullLODs);
+            var lods = metadata.LodGroup.GetLODs();
+
+            using (var pooledList = instantiatedLOD.GetComponentsInChildrenIntoPooledList<Renderer>(true))
+            {
+                var renderers = pooledList.Value.ToArray();
+                lods[CurrentLODLevelPromise].renderers = renderers;
+                if (loadedLODAmount == 1)
+                    CalculateCullRelativeHeight(renderers, defaultFOV, defaultLodBias, loadingDistance);
+            }
+
+            instantiatedLOD.transform.SetParent(metadata.LodGroup.transform);
+
+            metadata.LODAssets[CurrentLODLevelPromise] = lodAsset;
+
+            if (loadedLODAmount == 1)
             {
-                CurrentVisibleLOD?.Release();
-                CurrentVisibleLOD = CurrentLOD;
+                if (CurrentLODLevelPromise == 0)
+                {
+                    lods[0].screenRelativeTransitionHeight = metadata.CullRelativeHeight;
```

Comment:
Can you add some comments to these calculations? We will forget what's going on here in several months